Content/posts/toolbox-rewrite.md

```diff
@@ -0,0 +1,70 @@
+
+We have replaced the `mustach` wrapper with Hummingbird's [`swift-mustache`](https://github.com/hummingbird-project/swift-mustache), which is a Mustache rendering library written completely in Swift.
+
+We have also dropped the dependency on `ConsoleKit` and replaced it with Swift Argument Parser, which is now the recommended way to build command line tools in the Swift ecosystem.
+Swift Argument Parser doesn't support dynamic arguments out of the box, which are a fundamental feature of Toolbox templates.
+To work around this limitation, we have implemented [a solution](https://www.ackee.agency/blog/argumentparser-loading-dynamic-arguments) that relies on custom reflection, which allows us to dynamically generate the command line arguments based on the template manifest file.
+
+The new Toolbox is written in Swift 6, so we can use the latest language features and improvements, such as Swift Testing for unit tests and Swift Format for code linting.
+
+## New features and backwards compatibility
+
+All templates that worked with the old Toolbox, including [Vapor's official one](https://github.com/vapor/template) and custom ones, should continue to work with the new rewrite.
+
+The defining characteristic of a Vapor Toolbox template is the presence of a YAML manifest file, which lists the user input variables and specifies the Mustache template files that will be rendered using those values, and the overall structure of this file hasn't changed.
+
+However, we have added new functionalities to it, such as allowing nested variables for conditional rendering of files and directories, and the possibility to define a custom path to the manifest file via the new `--manifest` flag (by default it's `manifest.yml`, as before).
+
+We have also added a new `--verbose` flag to the `new` command, which when enabled will print output similar to the old Toolbox.
+The new default console output is much more concise.
+
+There's also a new **hidden** `--dump-variables` flag, which dumps the template variables as JSON.
+This can be useful when integrating the Toolbox with other tools, say for example a GUI or (spoiler alert!) a VS Code extension.
+
+> Note: The `--dump-variables` flag is experimental and unstable. Future releases may change its behavior or remove it.
+
+Despite the fact that the manifest file structure hasn't changed much and should be mostly backward compatible, there might be some minor changes with the flags, for example the short version of the `--template` flag has changed from `-T` to `-t`.
+
+This new version of the Toolbox has therefore been released as a major version bump and we don't guarantee full backwards compatibility.
+
+Take a look at the `--help` screen for the `new` command to see all the available flags and options.
+
+## Wrapping up
+
+The Vapor Toolbox rewrite is available as a new release on macOS (distributed via Homebrew as always) and Linux, where you'll have to build it from source.
```

Comment:
We may want to change this if we distribute via Homebrew on Linux as well.

Reply:
Agreed
```suggestion
The new Vapor Toolbox is available via Homebrew on macOS and Linux or can be compiled from source by cloning the GitHub repository and using the `Makefile` to install it.
```